pkg/util/tools/bijection.go

```diff
@@ -0,0 +1,93 @@
+package tools
+
+import (
+	"fmt"
+	"sync"
+)
+
+func NewBijectionMap[T1, T2 comparable]() *BijectionMap[T1, T2] {
+	return &BijectionMap[T1, T2]{
+		lock:       sync.RWMutex{},
+		forwardMap: map[T1]T2{},
+		reverseMap: map[T2]T1{},
+	}
+}
+
+type BijectionMap[T1, T2 comparable] struct {
+	lock sync.RWMutex
+
+	forwardMap map[T1]T2
+	reverseMap map[T2]T1
+}
+
+func (m *BijectionMap[T1, T2]) Lookup(key T1) (value T2, exists bool) {
```

Comment:
The words "forward", "backward", "reverse", etc are not very clear. What about calling them `T1ToT2` and `T2ToT1`? (This looks quite ugly, maybe rename the params to `X` and `Y` too so that it becomes `X2Y` and `Y2X`)